example/src/ThemeLoader.js

```diff
@@ -0,0 +1,70 @@
+import cssVarsPonyfill from 'css-vars-ponyfill';
+
+const THEME_URL_PARAM ='themeId';
+const THEME_CDN_URL   = './';
+
+class ThemeLoader {
+
+    static load() {
+
+        return new Promise(((resolve, reject) => {
+
+
+            /**
+             * Get the parameter `themeId` from the URL in case if it exists, and append
+             * it in a `link` element in the DOM
+             */
+            const applyThemeIfExists = () => {
+
+                const queryString = window.location.search.substr(1);
+
+                if (queryString === '') return;
+
+                queryString.split('&').forEach((item) => {
+
+                    const queryParam = item.split('=');
+
+                    if (queryParam[0] === THEME_URL_PARAM) {
+                        const styleElement= document.createElement('link');
+                        styleElement.type = 'text/css';
+                        styleElement.rel = 'stylesheet';
+                        styleElement.href = `${THEME_CDN_URL}${queryParam[1]}`;
+
+                        const head = document.getElementsByTagName('head')[0]
+                        head.appendChild(styleElement);
+                    }
+
+                });
+
+            };
+
+            /**
+             * Loads the CSS ponyfill.
+             *
+             * In case the property `onlyLegacy` is set to true, it will run only in legacy browsers like IE11.
+             * For test purposes, it's set to true, so can be tested on modern browsers.
+             */
+            const startCssVarsPonyfill = () => {
+                cssVarsPonyfill({
+                    onlyLegacy: false,
+                    onlyVars: true,
+                    watch: true,
+                    onComplete: function (cssText, node, url) {
+                        resolve();
+                    },
+                    onError: function (message, node, xhr, url) {
+                        console.log(message, node, xhr, url);
+                    },
+                });
+
+            };
+
+            startCssVarsPonyfill();
+            applyThemeIfExists();
+
+        }));
+    }
+}
+
+
+export default ThemeLoader;
```

Comment:
Ok, now it's better, but still some room for improvement:

1. I'd leave `startCssVarsPonyfill()` and `applyThemeIfExists()` methods as static in order not to put everything inside the `load()`
2. Both methods should return a `Promise`
3. Then load would look like:
```js
static load() {
        return Promise.all([
            ThemeLoader.applyThemeIfExists(),
            ThemeLoader.startCssVarsPonyfill(),
        ])
}
```

P.S. Both methods should do proper error handling (call `reject` in case of error, not `console.log`)

Reply:
Also probably, `ThemeLoader.applyThemeIfExists()` and `ThemeLoader.startCssVarsPonyfill()` should be called in a sequence one after another. Need to be testes

Reply:
I tested here and seems that the order doesn't affect the parsing